frontend/src/metabase/metadata/pages/DataModel/components/TablePicker/TablePicker.tsx

```diff
@@ -70,7 +70,7 @@ function Tree({
 
     if (
       !isExpanded({ databaseId: database.value.databaseId }) &&
-      database.value.databaseId !== databaseId
+      databaseId == null
```

Comment:
When db was not found, we automatically navigated to an existing one.
We want to show 404 error instead.